Objects/Converters/ConverterCSI/ConverterCSIShared/Models/Element2DAnalyticalResultConverter.cs

```diff
@@ -0,0 +1,238 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using ConverterCSIShared.Extensions;
+using CSiAPIv1;
+using Objects.Structural.Analysis;
+using Objects.Structural.CSI.Geometry;
+using Objects.Structural.Geometry;
+using Objects.Structural.Loading;
+using Objects.Structural.Results;
+using Speckle.Core.Models;
+
+namespace ConverterCSIShared.Models
+{
+  internal class Element2DAnalyticalResultConverter
+  {
+    private readonly Model speckleModel;
+    private readonly cSapModel sapModel;
+    private readonly Dictionary<string, Base> loadCombinationsAndCases;
+    private readonly bool sendForces;
+    private readonly bool sendStresses;
+    public Element2DAnalyticalResultConverter(
+      Model speckleModel,
+      cSapModel sapModel,
+      IEnumerable<LoadCombination> loadCombinations,
+      IEnumerable<LoadCase> loadCases,
+      bool sendForces,
+      bool sendStresses)
+    {
+      this.speckleModel = speckleModel;
+      this.sapModel = sapModel;
+      this.sapModel = sapModel;
+
+      this.loadCombinationsAndCases = new();
+      foreach (var combo in loadCombinations)
+      {
+        this.loadCombinationsAndCases.Add(combo.name, combo);
+      }
+      foreach (var loadCase in loadCases)
+      {
+        this.loadCombinationsAndCases.Add(loadCase.name, loadCase);
+      }
+
+      this.sendForces = sendForces;
+      this.sendStresses = sendStresses;
+    }
+
+    public void AnalyticalResultsToSpeckle()
+    {
+      foreach (Base element in speckleModel.elements)
+      {
+        if (element is not CSIElement2D element2D)
+        {
+          continue;
+        }
+
+        AnalyticalResults results = new()
+        {
+          resultsByLoadCombination = GetAnalysisResultsForElement2D(element2D).Cast<Result>().ToList()
+        };
+        element2D.AnalysisResults = results;
+      }
+    }
+
+    private ICollection<ResultSet2D> GetAnalysisResultsForElement2D(Element2D element2D)
+    {
+      int forceSuccess = -1;
+      int stressSuccess = -1;
+
+      int numberOfForceResults = 0;
+      string[] obj = Array.Empty<string>();
```

Comment:
Have we leaked the CSI API out into our own, these constant variable definitions look a bit painful

Reply:
Yeah I think we should discuss this point